flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -432,6 +434,19 @@ public Task(
 
         // finally, create the executing thread, but do not start it
         executingThread = new Thread(TASK_THREADS_GROUP, this, taskNameWithSubtask);
+
+        // Add CallerContext.
+        String callerContext =
+                "Flink_Task_"
+                        + "JobID_"
+                        + jobId
+                        + "_TaskName_"
+                        + taskInfo.getTaskName()
+                        + "_"
+                        + taskInfo.getAttemptNumber();
+
+        HadoopFsFactory.setCurrent(
+                tmConfig.getOptional(HadoopOptions.CALLER_CONTEXT_APP_ID).orElse(callerContext));
```

Comment:
Using `orElseGet` directly maybe better? 

If `CALLER_CONTEXT_APP_ID` exist, there's no need to construct the `callContext` var.